hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AsyncServerSocket.java

```diff
@@ -42,7 +44,7 @@ public abstract class AsyncServerSocket implements Closeable {
     protected final TpcLogger logger = TpcLoggerLocator.getLogger(getClass());
     protected final AtomicBoolean closed = new AtomicBoolean(false);
 
-    public AsyncServerSocket(){
+    protected AsyncServerSocket() {
```

Comment:
There is a socket option parity between `AsyncServerSocket` and `AsyncSocket`. Is this intentional? For example new added KEEP_* socket options is only available for `AsyncSocket`?